### PR TITLE
Cleanup category.py docstrings.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

For the better or (mostly) the worst, we support both bytes and str.
AnyStr is the union of both of them (per stdlib typing).

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by jklymak** [comment]
AnyStr seems pretty obscure though.  Unless there is some reason to steer people to byte strings is there any reason to be precise about the docstring?

**Comment by anntzer** [comment]
I'm fine either way.  (Or we can go for `Union[str, bytes]` too.)

File: lib/matplotlib/category.py [hunk 1]
```diff
@@ -116,14 +116,16 @@ def __init__(self, units_mapping):
         """
         Parameters
         -----------
-        units_mapping : Dict[str, int]
+        units_mapping : Dict[AnyStr, int]
```
Comment (timhoffm) [inline lib/matplotlib/category.py]: Usually we do not use type annotation syntax in the docstrings. I suppose this came in before we consistently adhered to numpydoc. I'd go for
```suggestion
        units_mapping : dict
            Mapping category names (str) to indices (int).
```

**Comment by anntzer** [comment]
went with @timhoffm's suggestion

**Comment by timhoffm** [comment]
Waiting to see if there are objections. Otherwise can self-merge after a day or so.

**Pull request merged**